### PR TITLE
fix for MarkDuplicatesSpark not writing output bam

### LINKED ISSUE #1448: MarkDuplicatesSpark produces an empty bam file when run on a stand alone spark
This was reported by @geetduggal in #1444.  

If you run MarkDuplicatesSpark using a local standalone spark cluster it can run to completion, but produces an empty bam file that has only a header.

```
./gatk-launch MarkDuplicatesSpark -I src/test/resources/large/CEUTrio.HiSeq.WGS.b37.NA12878.20.21.bam -O testoutput.bam -- --sparkRunner SPARK --sparkMaster spark://wm1b0-8ab:7077
```

produces an empty bam

### PR DESCRIPTION
**Pull request by lbergelson**
fixes #1448

MarkDuplicates spark was not writing an output bam when running with a standalone spark instance if the output path was a relative file path.
This fixes the problem by making any relative file paths into absolute file paths.
Added a check to see if no part files can be found so that errors like this will crash in the future instead of silently failing.

No tests are added because it's still unclear how to test errors that only occur in certain spark configurations.

`makeFilePathAbsolute()` should be redundant once #958 is finished

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java [hunk 1]
```diff
@@ -255,4 +263,12 @@ private static void mergeInto(OutputStream out, Path directory, Configuration co
...
         return parts;
     }
+
+    private static String makeFilePathAbsolute(String path){
+        if(BucketUtils.isCloudStorageUrl(path) || BucketUtils.isHadoopUrl(path)){
+            return path;
+        } else {
+            return new File(path).getAbsolutePath();
+        }
+    }
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java]: Is it possible to add a test case that failed before this patch, and passes now?

**Comment by droazen** [comment]
One comment, then feel free to merge (though watch out for merge conflicts with @tomwhite's just-merged Spark engine changes)

File: src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java (hunk 1 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java]: Is it possible?  Yes.  Do I know how to? Not yet..

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java]: How about at least calling `mergeInto()` from a test on a directory with no bam fragments, and confirming that it throws?

**Pull request merged**